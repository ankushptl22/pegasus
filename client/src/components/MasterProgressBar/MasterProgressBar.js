import React, { Component } from 'react';
import { Icon, Progress, Segment, Divider } from 'semantic-ui-react';
import './MasterProgressBar.less';

class MasterProgressBar extends Component {
  constructor(props) {
    super(props);
    //now set default vaule in case if it is not in props. 
      //this.state = { progressData: this.props.progressData };
     
  }

  render() {
console.log(this.props.progressData)
    const cmsBenchMarkJsx = (this.props.progressData.isCmsBenchMark) ? <Segment.Group className="mark-parent cms-bench-mark" style={{ left: `${this.props.progressData.cmsBenchMark}%` }}>
      <Segment textAlign="center" className="mark-child" > <Icon fitted name="sort up" size="big" className="master-icon sort-up" /></Segment>
      <Segment textAlign="center" className="mark-child" >{this.props.progressData.cmsBenchMark} % </Segment>
    </Segment.Group> : null;

    const RegistryAverageJsx = (this.props.progressData.isRegistryAverage) ? <Segment.Group className="mark-parent" style={{ left: `${this.props.progressData.registryAverage}%` }}>
      <Segment textAlign="center" className="mark-child" > {this.props.progressData.registryAverage} % </Segment>
      <Segment textAlign="center" className="mark-child" ><Icon fitted name="sort down" size="big" className="master-icon" /></Segment>
    </Segment.Group> : null;

    let difference;

    if (this.props.progressData.difference === null)
      difference = ((this.props.progressData.isCmsBenchMark) ? this.props.progressData.cmsBenchMark : this.props.progressData.registryAverage) - this.props.progressData.percentPerformance;
    else
      difference = this.props.progressData.difference;

    let progressColorCode = (difference >= 0) ? "progressbar progress-bar-success" : (Difference >= mean) ? "progressbar progress-bar-warning" : "progressbar progress-bar-danger";

    return (
      <Segment.Group horizontal className="no-box-shadow no-padding no-margin">
        <Segment basic textAlign="center" className="no-box-shadow no-border no-margin width-90 ">
          {RegistryAverageJsx}
          <Segment basic className='progress-segment' textAlign='left'>
            <Segment basic className="measure-performance-bar" style={{ left: `${this.props.progressData.percentPerformance}%` }} >I</Segment>
            <Progress className={progressColorCode} percent={this.props.progressData.percentPerformance} fitted color="black" size="tiny" />
          </Segment>
          {cmsBenchMarkJsx}
        </Segment>
        <Segment basic textAlign='left' className="no-box-shadow no-margin font-size-20 width-10 vertical-align-middle" >
          <label className={progressColorCode}>{this.props.progressData.percentPerformance} %</label>  </Segment>
      </Segment.Group>
    );
  }


}
MasterProgressBar.defaultProps = {
  progressData: {
    percentPerformanceText: "Achieved Performance",
    percentPerformance: 0,
    cmsBenchMark: 50,
    isCmsBenchMark: false,
    registryAverage: 50,
    isRegistryAverage: true,
    difference: 5,
    mean: 5
  }
};

export default MasterProgressBar;
