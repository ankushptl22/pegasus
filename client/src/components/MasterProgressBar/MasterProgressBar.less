.no-padding{
    padding:0px !important;
}
.no-margin{
    margin: 0px !important;

}
.no-margin{
    border:0px !important;
}

.no-box-shadow{
   box-shadow:none !important;
    }
    
.width-90{
    width:90% !important;
}

.width-10{
    width:10% !important;
}

.font-size-20{
    font-size:20px !important;
}

.vertical-align-middle{
    vertical-align:middle !important;
}
.mark-parent{
    width:62px !important;  
    border:0px !important; 
    box-shadow: none !important; 
    margin:0px !important;
}
.mark-child{
    width:62px !important;
    height:16px !important;    
    padding:0px !important;
    border:0px !important; 
    box-shadow: none !important; 
}

.master-icon{
    line-height:0.1 !important;
    height:11px !important;
}

.sort-up{
    line-height:0.5 !important;
}
 
.measure-performance-bar{
    font-size:12px !important;
    vertical-align:middle !important;
    width:2px !important;
    margin: 0px !important;
    padding: 0px !important;
    border:0px !important; 
    box-shadow: none !important; 
    position: absolute !important;
    top: -7px!important;
    font-weight: bold !important;
}
 

.progress-segment{
    height: 5px !important;
    padding:2px 0px !important;
    margin:0px !important;
}

.progressbar{
    height: 2px !important;
    margin:0px !important;
}
.progressbar .bar{
    border-radius: 0px !important;  
}