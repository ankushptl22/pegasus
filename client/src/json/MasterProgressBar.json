{
    "progressData": {
    "percentPerformanceText": "Achieved Performance",
    "percentPerformance": 66.05,
    "cmsBenchMark": 45,
    "isCmsBenchMark": true,
    "registryAverage": 82,
    "isRegistryAverage": true,
    "difference": 5,
    "mean": 5
}
}