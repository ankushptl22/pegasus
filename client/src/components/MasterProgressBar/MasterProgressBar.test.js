import React from 'react';
import { shallow } from 'enzyme';
import MasterProgressBar from './MasterProgressBar';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MasterProgressProps from '../../json/MasterProgressBar.json';

Enzyme.configure({ adapter: new Adapter() });

const progressData = MasterProgressProps.progressData;
describe('<MasterProgressBar />', () => {
  it('renders', () => {
    const wrapper = shallow(<MasterProgressBar  />);
    expect(wrapper).toMatchSnapshot();
  });

  it('mount component with default props', () => {
    const component = shallow(<MasterProgressBar  />);
    expect(component.exists()).toBe(true);
  });

  it('will check props', () => {
    const component = shallow(<MasterProgressBar progressData={MasterProgressProps} />).props();
    expect(wrapper).toBeDefined();
  });


});
