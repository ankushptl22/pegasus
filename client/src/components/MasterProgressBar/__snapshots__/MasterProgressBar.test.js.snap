// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MasterProgressBar /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MasterProgressBar
    progressData={
      Object {
        "cmsBenchMark": 50,
        "difference": 5,
        "isCmsBenchMark": false,
        "isRegistryAverage": true,
        "mean": 5,
        "percentPerformance": 0,
        "percentPerformanceText": "Achieved Performance",
        "registryAverage": 50,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Segment
          basic={true}
          className="no-box-shadow no-border no-margin width-90 "
          textAlign="center"
        >
          <SegmentGroup
            className="mark-parent"
            style={
              Object {
                "left": "50%",
              }
            }
          >
            <Segment
              className="mark-child"
              textAlign="center"
            >
               
              50
               % 
            </Segment>
            <Segment
              className="mark-child"
              textAlign="center"
            >
              <Icon
                as="i"
                className="master-icon"
                fitted={true}
                name="sort down"
                size="big"
              />
            </Segment>
          </SegmentGroup>
          <Segment
            basic={true}
            className="progress-segment"
            textAlign="left"
          >
            <Segment
              basic={true}
              className="measure-performance-bar"
              style={
                Object {
                  "left": "0%",
                }
              }
            >
              I
            </Segment>
            <Progress
              className="progressbar progress-bar-success"
              color="black"
              fitted={true}
              percent={0}
              size="tiny"
            />
          </Segment>
        </Segment>,
        <Segment
          basic={true}
          className="no-box-shadow no-margin font-size-20 width-10 vertical-align-middle"
          textAlign="left"
        >
          <label
            className="progressbar progress-bar-success"
          >
            0
             %
          </label>
            
        </Segment>,
      ],
      "className": "no-box-shadow no-padding no-margin",
      "horizontal": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "basic": true,
          "children": Array [
            <SegmentGroup
              className="mark-parent"
              style={
                Object {
                  "left": "50%",
                }
              }
            >
              <Segment
                className="mark-child"
                textAlign="center"
              >
                 
                50
                 % 
              </Segment>
              <Segment
                className="mark-child"
                textAlign="center"
              >
                <Icon
                  as="i"
                  className="master-icon"
                  fitted={true}
                  name="sort down"
                  size="big"
                />
              </Segment>
            </SegmentGroup>,
            <Segment
              basic={true}
              className="progress-segment"
              textAlign="left"
            >
              <Segment
                basic={true}
                className="measure-performance-bar"
                style={
                  Object {
                    "left": "0%",
                  }
                }
              >
                I
              </Segment>
              <Progress
                className="progressbar progress-bar-success"
                color="black"
                fitted={true}
                percent={0}
                size="tiny"
              />
            </Segment>,
            null,
          ],
          "className": "no-box-shadow no-border no-margin width-90 ",
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Segment
                  className="mark-child"
                  textAlign="center"
                >
                   
                  50
                   % 
                </Segment>,
                <Segment
                  className="mark-child"
                  textAlign="center"
                >
                  <Icon
                    as="i"
                    className="master-icon"
                    fitted={true}
                    name="sort down"
                    size="big"
                  />
                </Segment>,
              ],
              "className": "mark-parent",
              "style": Object {
                "left": "50%",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    " ",
                    50,
                    " % ",
                  ],
                  "className": "mark-child",
                  "textAlign": "center",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  50,
                  " % ",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Icon
                    as="i"
                    className="master-icon"
                    fitted={true}
                    name="sort down"
                    size="big"
                  />,
                  "className": "mark-child",
                  "textAlign": "center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "as": "i",
                    "className": "master-icon",
                    "fitted": true,
                    "name": "sort down",
                    "size": "big",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "basic": true,
              "children": Array [
                <Segment
                  basic={true}
                  className="measure-performance-bar"
                  style={
                    Object {
                      "left": "0%",
                    }
                  }
                >
                  I
                </Segment>,
                <Progress
                  className="progressbar progress-bar-success"
                  color="black"
                  fitted={true}
                  percent={0}
                  size="tiny"
                />,
              ],
              "className": "progress-segment",
              "textAlign": "left",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "basic": true,
                  "children": "I",
                  "className": "measure-performance-bar",
                  "style": Object {
                    "left": "0%",
                  },
                },
                "ref": null,
                "rendered": "I",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "progressbar progress-bar-success",
                  "color": "black",
                  "fitted": true,
                  "percent": 0,
                  "size": "tiny",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "basic": true,
          "children": Array [
            <label
              className="progressbar progress-bar-success"
            >
              0
               %
            </label>,
            "  ",
          ],
          "className": "no-box-shadow no-margin font-size-20 width-10 vertical-align-middle",
          "textAlign": "left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                0,
                " %",
              ],
              "className": "progressbar progress-bar-success",
            },
            "ref": null,
            "rendered": Array [
              0,
              " %",
            ],
            "type": "label",
          },
          "  ",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Segment
            basic={true}
            className="no-box-shadow no-border no-margin width-90 "
            textAlign="center"
          >
            <SegmentGroup
              className="mark-parent"
              style={
                Object {
                  "left": "50%",
                }
              }
            >
              <Segment
                className="mark-child"
                textAlign="center"
              >
                 
                50
                 % 
              </Segment>
              <Segment
                className="mark-child"
                textAlign="center"
              >
                <Icon
                  as="i"
                  className="master-icon"
                  fitted={true}
                  name="sort down"
                  size="big"
                />
              </Segment>
            </SegmentGroup>
            <Segment
              basic={true}
              className="progress-segment"
              textAlign="left"
            >
              <Segment
                basic={true}
                className="measure-performance-bar"
                style={
                  Object {
                    "left": "0%",
                  }
                }
              >
                I
              </Segment>
              <Progress
                className="progressbar progress-bar-success"
                color="black"
                fitted={true}
                percent={0}
                size="tiny"
              />
            </Segment>
          </Segment>,
          <Segment
            basic={true}
            className="no-box-shadow no-margin font-size-20 width-10 vertical-align-middle"
            textAlign="left"
          >
            <label
              className="progressbar progress-bar-success"
            >
              0
               %
            </label>
              
          </Segment>,
        ],
        "className": "no-box-shadow no-padding no-margin",
        "horizontal": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "basic": true,
            "children": Array [
              <SegmentGroup
                className="mark-parent"
                style={
                  Object {
                    "left": "50%",
                  }
                }
              >
                <Segment
                  className="mark-child"
                  textAlign="center"
                >
                   
                  50
                   % 
                </Segment>
                <Segment
                  className="mark-child"
                  textAlign="center"
                >
                  <Icon
                    as="i"
                    className="master-icon"
                    fitted={true}
                    name="sort down"
                    size="big"
                  />
                </Segment>
              </SegmentGroup>,
              <Segment
                basic={true}
                className="progress-segment"
                textAlign="left"
              >
                <Segment
                  basic={true}
                  className="measure-performance-bar"
                  style={
                    Object {
                      "left": "0%",
                    }
                  }
                >
                  I
                </Segment>
                <Progress
                  className="progressbar progress-bar-success"
                  color="black"
                  fitted={true}
                  percent={0}
                  size="tiny"
                />
              </Segment>,
              null,
            ],
            "className": "no-box-shadow no-border no-margin width-90 ",
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <Segment
                    className="mark-child"
                    textAlign="center"
                  >
                     
                    50
                     % 
                  </Segment>,
                  <Segment
                    className="mark-child"
                    textAlign="center"
                  >
                    <Icon
                      as="i"
                      className="master-icon"
                      fitted={true}
                      name="sort down"
                      size="big"
                    />
                  </Segment>,
                ],
                "className": "mark-parent",
                "style": Object {
                  "left": "50%",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      " ",
                      50,
                      " % ",
                    ],
                    "className": "mark-child",
                    "textAlign": "center",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    50,
                    " % ",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <Icon
                      as="i"
                      className="master-icon"
                      fitted={true}
                      name="sort down"
                      size="big"
                    />,
                    "className": "mark-child",
                    "textAlign": "center",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "as": "i",
                      "className": "master-icon",
                      "fitted": true,
                      "name": "sort down",
                      "size": "big",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "basic": true,
                "children": Array [
                  <Segment
                    basic={true}
                    className="measure-performance-bar"
                    style={
                      Object {
                        "left": "0%",
                      }
                    }
                  >
                    I
                  </Segment>,
                  <Progress
                    className="progressbar progress-bar-success"
                    color="black"
                    fitted={true}
                    percent={0}
                    size="tiny"
                  />,
                ],
                "className": "progress-segment",
                "textAlign": "left",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "basic": true,
                    "children": "I",
                    "className": "measure-performance-bar",
                    "style": Object {
                      "left": "0%",
                    },
                  },
                  "ref": null,
                  "rendered": "I",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "progressbar progress-bar-success",
                    "color": "black",
                    "fitted": true,
                    "percent": 0,
                    "size": "tiny",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "basic": true,
            "children": Array [
              <label
                className="progressbar progress-bar-success"
              >
                0
                 %
              </label>,
              "  ",
            ],
            "className": "no-box-shadow no-margin font-size-20 width-10 vertical-align-middle",
            "textAlign": "left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " %",
                ],
                "className": "progressbar progress-bar-success",
              },
              "ref": null,
              "rendered": Array [
                0,
                " %",
              ],
              "type": "label",
            },
            "  ",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
